srcs/parsing/parser.c

```diff
@@ -2,14 +2,57 @@
 
 void	parse(char *line)
 {
-	// skip space
+	t_minishell	*minishell;
+	int	i;
+
+	minishell = get_minishell(NULL);
+	minishell->instructions = ft_split(line, PIPE);
+	minishell->nb_cmd = get_split_len(minishell->instructions);
+	minishell->cmd_table = init_cmd_table(minishell->nb_cmd);
+	i = 0;
+	while (i < minishell->nb_cmd)
+	{
+		minishell->cmd_table[i]->cmd_arg = tokenise(minishell->instructions[i], SPACE);
+		DEBUG(print_tokens("%s\n", minishell->cmd_table[i]->cmd_arg);) // to test !!!
+		i++;
+	}
+
+	// make token struct with key/type etc.
 	// get command
 	// skip space
 	// until it doesn't start by '-' + alphanumerical (until stop alphanumerical, get flag
 	// skip space
 	// check redir
 	// check quote
 	// save arg
-	printf("%s\n", line);
+	free_split(minishell->instructions); 
 	return ;
-}
\ No newline at end of file
+}
+
+t_cmd_table	**init_cmd_table(int nb_cmd)
+{
+	t_cmd_table	**cmd_table;

```

Comment:
I find this a bit confusing, you call it cmd_table in singular but then it's actually an array, in plural. Or is the "table" part that makes it plural? If yes then I'd call the struct t_cmd instead of t_cmd_table. This way it seems that a cmd_table is a single element

Reply:
then t_cmd_table (one element definition) and cmd_tables (for the element we create ?) ?

Reply:
ah ok, so table is singular? then I'd call the array cmd_tables yes. Although now I'm finding it a bit weird that a table is singular 😅

Reply:
Well, the idea is you create a t_cmd_table (singular), it's a singular object/element. But then you create an array of cmd_tables (so plural). I find that makes sense.

Reply:
I need to try all that out to really get how it works. Maybe my t_token tokens also need to be a singular *

Reply:
okido